src/prefect/environments/storage/webhook.py

```diff
@@ -0,0 +1,404 @@
+import cloudpickle
+import os
+import string
+import warnings
+
+from collections.abc import Mapping
+from copy import deepcopy
+from typing import TYPE_CHECKING, Any, Dict, Iterator, List, Optional
+
+from requests import Session
+from requests.adapters import HTTPAdapter
+from requests.models import Response
+from requests.packages.urllib3.util.retry import Retry
+
+from prefect.client import Secret
+from prefect.environments.storage import Storage
+from prefect.utilities.storage import extract_flow_from_file
+
+if TYPE_CHECKING:
+    from prefect.core.flow import Flow
+
+
+class _SecretMapping(Mapping):
+    """
+    Magic mapping. When `__getitem__()` is called
+    to look for the value of a `key`, this mapping will
+    try to resolve it from the environment and, if not found
+    there, from `prefect` secrets.
+
+    Raises:
+        - RuntimeError if the key you're looking for isn't found
+    """
+
+    def __getitem__(self, key: str) -> str:
+        out = os.getenv(key, None)  # type: ignore
+        if out is None:
+            try:
+                out = Secret(key).get()
+            except ValueError:
+                msg = (
+                    "Template value '{}' does not refer to an "
+                    "environment variable or Prefect secret."
+                )
+                raise RuntimeError(msg.format(key))
+        return out  # type: ignore
+
+    def __iter__(self) -> Iterator[Any]:
+        return iter([])
+
+    def __len__(self) -> int:
+        return 0
+
+
+_mapping = _SecretMapping()
+
+
+def _render_dict(input_dict: Dict[str, Any]) -> Dict[str, Any]:
+    """
+    Replace string elements in a dictionary with environment variables
+    or Prefect secrets.
+
+    This method will look through all string values (not keys) of a dictionary,
+    recursively, for template string. When it encounters a string like
+    `${abc}`, it will first try to replace it with the value of environment
+    variable `abc`. If environment variable `abc` is not defined, it will try
+    to replace it with `prefect` secret `abc`. If that is also not found,
+    raises a `KeyError`.
+
+    Args:
+        - input_dict (Dict[str, Any]): A dictionary that may contain
+            template strings.
+
+    Returns:
+        - A dictionary with template strings replaced by the literal values of
+            environment variables or Prefect secrets
+
+    Raises:
+        - RuntimeError if any template value cannot be found in the environment
+            or `prefect` secrets
+    """
+    input_dict = deepcopy(input_dict)
```

Comment:
I don't believe deepcopy is necessary here, since you're building a new dict.

```suggestion
```

Reply:
oh sure! Fine with me if you want to drop it. I just always do that out of an abundance of caution. I expect these dicts to be so small that the performance and memory hit from doing that is negligble. I'm ok with removing it.

Reply:
removed in https://github.com/PrefectHQ/prefect/pull/3000/commits/921e387d26e624b248de962e3fda9b5a54f1db91. I decided not to commit the suggestion in the browser, since I also needed to remove `from copy import deepcopy` and didn't want to waste CI resources by making this change in two commits